x-pack/plugins/rule_registry/common/field_map/runtime_type_from_fieldmap.ts

```diff
@@ -77,7 +77,7 @@ type CastSingle<T extends t.Type<any>> = t.Type<
 >;
 
 const createCastArrayRt = <T extends t.Type<any>>(type: T): CastArray<T> => {
-  const union = t.union([type, t.array(type)]);
+  const union = t.union([type, t.array(t.union([type, t.nullType]))]);
```

Comment:
This change is related to the values that are saved in `kibana.alert.evaluation.values`, it is a mixture of the specified type (`scaled_float`) and `null`